packages/react-storage/src/components/StorageBrowser/subcomponents/PaginationControl/PaginationControl.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { classNames } from '@aws-amplify/ui';
+import { NavElementProps } from '@aws-amplify/ui-react/internal';
+import { useElement } from '../../context/elements';
+import { NextButton } from './NextButton';
+import { PreviousButton } from './PreviousButton';
+import { CurrentPage } from './CurrentPage';
+import { Item } from './Item';
+
+const PaginationControlPrimitive = <T extends NavElementProps>({
+  ariaLabel = 'Pagination',
+  className,
+  children,
+  ...rest
+}: T): JSX.Element => {
+  const Nav = useElement('Nav');
+
+  return (
+    <Nav
+      {...rest}
+      aria-label={ariaLabel}
+      className={classNames(
+        'amplify-storagebrowser-pagination-control',
+        className
+      )}
+    >
+      <ol className={``}>{children}</ol>
```

Comment:
Probably this ol item should be a subcomponent? I wasn't sure if this was getting TOO intricate, but this is how we markup our Pagination UI component (and how it commonly is marked up: a nav element wrapping a list)